### ISSUE TITLE
Improvements in line breaks

### ISSUE THREAD

**Issue by murdos** [eemeli/prettier-plugin-properties#15]
Using version 0.3.0, with `printWidth: 140` and `keySeparator: '='` configuration.

I'm not sure what rules could be inferred, but on these examples the properties before transformation are easier to read.

### Before
```properties
sonar.cpd.exclusions=\
  src/main/java/tech/jhipster/lite/error/infrastructure/primary/ArgumentsReplacer.java,\
  src/main/java/tech/jhipster/lite/module/domain/file/ArgumentsReplacer.java
```
### After
```properties
sonar.cpd.exclusions=src/main/java/tech/jhipster/lite/error/infrastructure/primary/ArgumentsReplacer.\
  java,src/main/java/tech/jhipster/lite/module/domain/file/ArgumentsReplacer.java
```

### Before
```properties
sonar.exclusions=\
  src/main/webapp/main.ts, \
  src/main/webapp/app/main.ts, \
  src/main/webapp/content/**/*.*, \
  src/main/webapp/i18n/*.js, \
  target/classes/static/**/*.*, \
  src/main/resources/**, \
  src/main/webapp/app/router/index.ts, \
  src/main/glyph/css/**, \
  src/main/webapp/app/common/primary/applicationlistener/WindowApplicationListener.ts
```

### After
```properties
sonar.exclusions=src/main/webapp/main.ts, src/main/webapp/app/main.ts, src/main/webapp/content/**/*.*, src/main/webapp/i18n/*.js, \
  target/classes/static/**/*.*, src/main/resources/**, src/main/webapp/app/router/index.ts, src/main/glyph/css/**, \
  src/main/webapp/app/common/primary/applicationlistener/WindowApplicationListener.ts
```

### Before
```properties
assertion-error.STRING_TOO_LONG.detail=\
  Content of {{ field }} is too long, must be less than {{ maxLength }} character(s) (was {{ currentLength }})
```

### After
```properties
assertion-error.STRING_TOO_LONG.detail=Content of {{ field }} is too long, must be less than {{ maxLength }} character(s) (was {{ \
  currentLength }})
```

**Comment by eemeli** [comment]
Yeah, those are hard. One improvement that they do suggest to me is that when a value does not fit on the same line with the key, it might make sense to always (?) start it on the next line instead, using an escaped newline.

Or are there cases where that would be worse choice than the current one?

**Comment by murdos** [comment]
Yes, always starting on the next line when a value does not fit on the same line with the key sounds like a good and safe idea.

I was also thinking about other heuristics, but I'm not sure they always apply:
- split to a new line after each comma `,`
- avoid splitting a "word" around a dot `.` if there's no space after of before the dot. This should avoid split e.g. 
  - `src/main/webapp/app/router/index.ts` to `src/main/webapp/app/router/index.` and `ts`. 
  - or `https://github.com/eemeli/prettier-plugin-properties` to `https://github.` and ` com/eemeli/prettier-plugin-properties`

**Comment by eemeli** [comment]
> split to a new line after each comma `,`

This would not really work for property values that are e.g. localizable text.

> avoid splitting a "word" around a dot `.` if there's no space after of before the dot.

In practice that would probably mean dropping [this line](https://github.com/eemeli/dot-properties/blob/fe49d2d779528a499c42bf102d04887d4374c2f3/lib/stringify.js#L62), which should be pretty safe.

Or maybe replacing it with `case ',':`?

Note that the changes under consideration are required in [`dot-properties`](https://github.com/eemeli/dot-properties), rather than this repo.

**Comment by eemeli** [comment]
This was fixed previously upstream.
